tests/Agent/IntegrationTests/IntegrationTestHelpers/AgentLogBase.cs

```diff
@@ -115,8 +115,8 @@ public string GetCrossProcessId(TimeSpan? timeoutOrZero = null)
 
         public string GetReportingAppLink(TimeSpan? timeoutOrZero = null)
         {
-            var match = WaitForLogLine(AgentReportingToLogLineRegex, timeoutOrZero);
-            return match.Groups[1].Value;
+            var match = TryGetLogLine(AgentReportingToLogLineRegex);
```

Comment:
Did the previous call to WaitForLogLine throw an exception previously? Just wondering if we should throw an exception, or return null like you have here?

Reply:
`WaitForLogLine()` does throw an exception after the timeout expires - with the default timeout being 0. 

I might be able to just revert this particular change, though, as I'm thinking now that it was the logging in that I removed from `WaitForLogLines()` that was really causing the exceptions when we initially upgraded to 2.9.0.

Reply:
I'm not sure I follow -- why was logging here the only place triggering the exception?

Reply:
I think that it was some of the contexts under which the WaitForLogLines method was called, that triggered the problem.

Reply:
The crux of the issue seems to be trying to write log lines (like `WaitForLogLines()` did prior to this PR) after the test execution is over (i.e., while evaluating the results in a `[Fact]` method). 

I'm refactoring this whole chunk of code to use `TryGetLogLine()` instead and throw an exception if not found - that should match the previous behavior.